### ISSUE TITLE
comments-indentation: comment not indented like content should issue an error

### ISSUE THREAD

**Issue by pmalek-sumo** [adrienverge/yamllint#375]
I believe the following yaml: 

```yaml
## Random stuff

stuff:
  ## uncomment for the debug mode
  # debug: true
```

should cause an error like this:

```
  4:3       error    comment not indented like content  (comments-indentation)
```

Config used:

```yaml
extends: relaxed

rules:
    comments: enable
    comments-indentation: enable

    line-length:
        max: 140
        level: warning

    indentation:
        level: warning
        indent-sequences: consistent
```

**Comment by adrienverge** [comment]
I guess you meant "*should **not** cause an error like this*"?

This is probably a very subjective topic, depending on how you "see" and "feel" YAML.

I kind of agree with what you say. However, some others users would dislike such a change, because it would mean this wrongly-indented comment wouldn't be reported:
```yaml
todo list:
  item 1:
  item 2: ✓
  item 3:
    # item 4:
  item 5:
  item 6: ✓
```

**Comment by pmalek-sumo** [comment]
> I guess you meant "_should **not** cause an error like this_"?

Yup, corrected.

This kinda makes sense. Maybe this is specific for [our use case ](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/values.yaml) where we comment our yaml files extensively to make users aware what are they configuring.

Specifically we use `##` to indicate comments about below commented code (which is optional) to describe what it would configure in case it wasn't uncommented.

Maybe there's a way to separate this with what you described? OTOH they are both comments and `yamllint`'s config mentions  specifically about `comments-indentation` nothing else.

**Comment by Vanav** [comment]
Related: #141, #384.
